packages/core/src/types/oas2.ts

```diff
@@ -330,32 +320,48 @@ const SecurityScheme: NodeType = {
     scopes: { type: 'object', additionalProperties: { type: 'string' } },
   },
   required(value) {
-    if (!value?.type) {
-      return ['type'];
+    switch (value?.type) {
+      case 'apiKey':
+        return ['type', 'name', 'in'];
+      case 'oauth2':
+        switch (value?.flow) {
+          case 'implicit':
+            return ['type', 'flow', 'authorizationUrl', 'scopes'];
+          case 'accessCode':
+            return ['type', 'flow', 'authorizationUrl', 'tokenUrl', 'scopes'];
+          case 'application':
+          case 'password':
+            return ['type', 'flow', 'tokenUrl', 'scopes'];
+          default:
+            return ['type', 'flow', 'scopes'];
+        }
+      default:
+        return ['type'];
     }
-
-    if (value.type === 'apiKey') {
-      return ['type', 'name', 'in'];
-    } else if (value.type === 'http') {
-      return ['type', 'scheme'];
-    } else if (value.type === 'oauth2') {
-      if (!value?.flow) {
-        return ['type', 'flow'];
-      } else if (value.flow === 'implicit') {
-        return ['type', 'flow', 'authorizationUrl'];
-      } else if (value.flow === 'accessCode') {
-        return ['type', 'flow', 'authorizationUrl', 'tokenUrl'];
-      } else if (value.flow === 'application') {
-        return ['type', 'flow', 'tokenUrl'];
-      } else if (value.flow === 'password') {
-        return ['type', 'flow', 'tokenUrl'];
-      } else {
-        return ['type', 'flow'];
-      }
+  },
+  allowed(value) {
+    switch (value?.type) {
+      case 'basic':
+        return ['type', 'description'];
+      case 'apiKey':
+        return ['type', 'name', 'in', 'description'];
+      case 'oauth2':
+        switch (value?.flow) {
+          case 'implicit':
+            return ['type', 'flow', 'authorizationUrl', 'description', 'scopes'];
+          case 'accessCode':
+            return ['type', 'flow', 'authorizationUrl', 'tokenUrl', 'description', 'scopes'];
+          case 'application':
+          case 'password':
+            return ['type', 'flow', 'tokenUrl', 'description', 'scopes'];
+          default:
+            return ['type', 'authorizationUrl', 'flow', 'scopes', 'description'];
```

Comment:
I would allow everything until flow is not defined, so add `tokenUrl` here too.
```suggestion
            return ['type', 'authorizationUrl', 'tokenUrl', 'flow', 'scopes', 'description'];
```